library/Vanilla/DateFilterSchema.php

```diff
@@ -181,26 +221,88 @@ public function validate($data, $sparse = false) {
      *
      * @param mixed $value The value to validate.
      * @param ValidationField $field The validation results to add.
-     * @return string|Invalid Returns the valid string or **null** if validation fails.
+     * @return string|false Returns the valid string or **null** if validation fails.
      */
-    private function validateDateFilter($value, ValidationField $field) {
+    private function stringToSchema($value, ValidationField $field) {
         $result = Invalid::value();
 
-        if (is_string($value)) {
-            $first = substr($value, 0, 1);
-            if (in_array($first, $this->rangeOpen)) {
-                $last = substr($value, -1, 1);
-                if (in_array($last, $this->rangeClose) && strlen($value) > 2) {
-                    $dates = substr($value, 1, -1);
-                    $result = $this->parseRange($dates, $first, $last, $field);
+        $escapedOpen = '['.implode('', array_map('preg_quote', $this->rangeOpen)).']';
+        $escapedClose = '['.implode('', array_map('preg_quote', $this->rangeClose)).']';
+
+        // Sort the operators so that the matches occur on the longest operators first.
+        $sortedSimpleOperators = $this->simpleOperators;
+        usort($sortedSimpleOperators, function($a, $b) {
+            if (strlen($a) > strlen($b)) {
+                return -1;
+            } else if (strlen($a) < strlen($b)) {
+                return 1;
+            }
+            return 0;
+        });
+        $simpleOperators = '(?<op>'.implode('|', array_map('preg_quote', $sortedSimpleOperators)).'|)';
```

Comment:
This is going to give you something like: `(?<op>\<\=|\>\=|\<|\>|\=|)`. Is the trailing `|` there actually necessary?

Reply:
yes

Reply:
2017-01-01 is valid since no operator defaults to '='